pinot-core/src/main/java/com/linkedin/pinot/core/operator/transform/function/converter/TransformConverter.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.operator.transform.function.converter;
+
+import javax.annotation.Nonnull;
+
+
+/**
+ * Interface for transform converter to batch convert array of input values and store the results into the output array.
+ *
+ * @param <I> Type of input array
+ * @param <O> Type of output array
+ */
+public interface TransformConverter<I, O> {
```

Comment:
Seems like this does not need to be tied to transform module? If so, can we take it out and rename it something like DataTransformer?

Reply:
Done